adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/AmqpContext.java

```diff
@@ -0,0 +1,72 @@
+package org.eclipse.hono.adapter.amqp;
+
+import org.apache.qpid.proton.message.Message;
+import org.eclipse.hono.util.AmqpErrorException;
+import org.eclipse.hono.util.MessageHelper;
+import org.eclipse.hono.util.ResourceIdentifier;
+
+import io.vertx.core.buffer.Buffer;
+import io.vertx.proton.ProtonDelivery;
+
+/**
+ * A class that contains context information used by the AMQP Adapter when uploading messages to Hono.
+ */
+public class AmqpContext {
+
+    private final ProtonDelivery delivery;
+    private final Message message;
+    private final ResourceIdentifier resource;
+
+    AmqpContext(final ProtonDelivery delivery, final Message message, final ResourceIdentifier resource) {
+        this.delivery = delivery;
+        this.message = message;
+        this.resource = resource;
+    }
+
+    /**
+     * Gets the body of the AMQP 1.0 message.
+     * 
+     * @return The body of the AMQP 1.0 message as a buffer object.
+     */
+    Buffer getMessagePayload() {
+        return MessageHelper.getPayload(message);
+    }
+
+    /**
+     * Gets the content type of the AMQP 1.0 message.
+     * 
+     * @return The content type of the AMQP 1.0 message.
+     */
+    String getMessageContentType() {
+        return message.getContentType();
+    }
+
+    ProtonDelivery delivery() {
```

Comment:
are these methods intentionally declared package private?

Reply:
yes, at least for now.

Reply:
Ok, a JavaDoc would still be helpful, I guess :-)